docs/docs/end-to-end-testing.md

```diff
@@ -49,10 +51,125 @@ If you want to run Cypress in Continuous Integration (CI) you have to use `cypre
+
+```js:title=cypress/e2e/a11y.test.js
+/// <reference types="Cypress" />
+
+describe("Accessibility checks", () => {
+  beforeEach(() => {
+    cy.visit("/")
+    cy.injectAxe()
+    cy.wait(500)
+  })
+  it("Has no detectable a11y violations on load", () => {
+    cy.checkA11y()
+  })
+})
+```
+
+You can run `test:e2e` to run the test. If you already have the development server open you can run `cy:open` instead.
+
+One thing to keep in mind is that you can't always see the exact error message from the sidebar (command log). For that, you have to open the developer console and find the message in the output. You can see how an accessibility error looks in the [cypress-axe GitHub page](https://github.com/avanslaars/cypress-axe#output).
+
+You don't have to use the `checkA11y` method only on page load. For example, you can perform a click on a button and check again. This is especially useful if that button opens a modal or a mobile menu for example.
+
+The following test is for the [gatsby-default-starter](https://github.com/gatsbyjs/gatsby-starter-default). You visit the homepage and you search for the link that goes to page 2 with the `getByText` command. Then you perform a click and check for accessibility errors in the second page.
+
+```js:title=cypress/e2e/a11y.test.js
+/// <reference types="Cypress" />
+
+describe("Accessibility checks", () => {
+  beforeEach(() => {
+    cy.visit("/")
+    cy.injectAxe()
+    cy.wait(500)
+  })
+  it("Has no detectable a11y violations on load", () => {
+    cy.checkA11y()
+  })
+  // highlight-start
+  it("Navigates to page 2 and checks for accessibility violations", () => {
+    cy.getByText(/go to page 2/i)
+      .click()
+      .checkA11y()
+  })
+  // highlight-end
+})
+```
+
+You can also make [assertions](https://docs.cypress.io/guides/core-concepts/introduction-to-cypress.html#Assertions) with the [should command](https://docs.cypress.io/api/commands/should.html#Syntax).
+
+In `gatsby-default-starter` homepage you can write another test that focuses on the footer link. Then you make some assertions on the item that's currently focused:
+
+```js:title=cypress/e2e/a11y.test.js
+/// <reference types="Cypress" />
+
+describe("Accessibility checks", () => {
+  beforeEach(() => {
+    cy.visit("/")
+    cy.injectAxe()
+    cy.wait(500)
+  })
+  it("Has no detectable a11y violations on load", () => {
+    cy.checkA11y()
+  })
+  it("Navigates to page 2 and checks for accessibility violations", () => {
+    cy.getByText(/go to page 2/i)
+      .click()
+      .checkA11y()
+  })
+  // highlight-start
+  it("Checks if footer link is focusable and has the correct attributes", () => {
+    cy.getAllByText("Gatsby")
+      .focus()
+      .checkA11y()
+
+    cy.focused()
+      .should("have.text", "Gatsby")
+      .should("have.attr", "href", "https://www.gatsbyjs.org")
+      .should("have.css", "outline-width", "1px")
```

Comment:
This line fails when I run it with `cypress run` (it pass with `cypress open`). It says the `outline-width` is `5px` for some reason.

I will change it to `.should("not.have.css", "outline-width", "0px")` to ensure that at least there's an outline.